crates/zkvm/zkvm/src/verifier.rs

```diff
@@ -0,0 +1,54 @@
+use arbitrary::Arbitrary;
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+
+use crate::Proof;
+
+/// A trait implemented by a verifier to decode and verify the proof generated by the prover
+/// ("host").
+pub trait ZkVmVerifier {
+    /// Verifies the proof generated by the prover against the `program_id`.
+    fn verify(verification_key: &VerificationKey, proof: &Proof) -> anyhow::Result<()>;
+
+    /// Verifies the proof generated by the prover against the given `program_id` and
+    /// `public_params`.
+    fn verify_with_public_params<T: Serialize + DeserializeOwned>(
+        verification_key: &VerificationKey,
+        public_params: T,
+        proof: &Proof,
+    ) -> anyhow::Result<()>;
+
+    fn verify_groth16(
+        proof: &Proof,
+        verification_key: &[u8],
+        public_params_raw: &[u8],
+    ) -> anyhow::Result<()>;
+
+    /// Extracts the public output from the given proof using standard `serde`
+    /// serialization/deserialization.
+    fn extract_serde_public_output<T: Serialize + DeserializeOwned>(
+        proof: &Proof,
+    ) -> anyhow::Result<T>;
```

Comment:
Same thing here as with the last round, I didn't pick this up.  `bincode` is the actual format, serde is just the framework.

Reply:
Yup. SP1 is using `bincode` while Risc0 has it's own serde compatible de/serializer.